src/server/scripts/EasternKingdoms/ZulAman/instance_zulaman.cpp

```diff
@@ -152,6 +153,51 @@ class instance_zulaman : public InstanceMapScript
                 }
             }
 
+            void OnUnitDeath(Unit* unit) override
+            {
+                InstanceScript::OnUnitDeath(unit);
+
+                Creature* creature = unit->ToCreature();
+                if (!creature)
+                    return;
+
+                if (creature->HasStringId("NalorakkWave1"))
+                {
+                    ++killedUnitInWaveCounter[0];
+
+                    if (killedUnitInWaveCounter[0] == 3)
```

Comment:
is respawn/reset/wipe handling necessary? meaning, do you need to reset this counter to 0 somewhere in case of any of these situations?